packages/client/utils/JiraServerClientManager.ts

```diff
@@ -0,0 +1,52 @@
+import Atmosphere from '../Atmosphere'
+import {MenuMutationProps} from '../hooks/useMutationProps'
+import AddTeamMemberIntegrationAuthMutation from '../mutations/AddTeamMemberIntegrationAuthMutation'
+import getOAuthPopupFeatures from './getOAuthPopupFeatures'
+import CreateOAuth1AuthorizeUrlMutation from '../mutations/CreateOAuth1AuthorizeUrlMutation'
+class JiraServerClientManager {
+  static SCOPES = 'read_api'
+  static openOAuth(
+    atmosphere: Atmosphere,
+    providerId: string,
+    teamId: string,
+    mutationProps: MenuMutationProps
+  ) {
+    const {submitting, onError, onCompleted, submitMutation} = mutationProps
+
+    // Open synchronously because of Safari
+    const popup = window.open(
+      '',
+      'OAuth',
+      getOAuthPopupFeatures({width: 500, height: 750, top: 56})
+    )
+
+    const onUrlCompleted = (result: any) => {
+      const uri = result.createOAuth1AuthorizeUrl.url
+      window.open(
+        uri,
+        'OAuth',
+        getOAuthPopupFeatures({width: 500, height: 750, top: 56})
+      )
```

Comment:
-1 if the popup is already opened, we should just change the URL `popup.location.replace(uri);` or `popup.location.href = uri`